include/oneapi/dpl/internal/reduce_by_segment_impl.h

```diff
@@ -138,6 +167,414 @@ reduce_by_segment_impl(Policy&& policy, InputIterator1 first1, InputIterator1 la
 }
 
 #if _ONEDPL_BACKEND_SYCL
+
+template <typename... Name>
+class __sycl_segmented_reduce_kernel1;
+template <typename... Name>
+class __sycl_segmented_reduce_kernel2;
+template <typename... Name>
+class __sycl_segmented_reduce_kernel3;
+template <typename... Name>
+class __sycl_segmented_reduce_kernel4;
+
+namespace
+{
+template <typename... _Name>
+using _KernelName1 = __sycl_segmented_reduce_kernel1<_Name...>;
+template <typename... _Name>
+using _KernelName2 = __sycl_segmented_reduce_kernel2<_Name...>;
+template <typename... _Name>
+using _KernelName3 = __sycl_segmented_reduce_kernel3<_Name...>;
+template <typename... _Name>
+using _KernelName4 = __sycl_segmented_reduce_kernel4<_Name...>;
+} // namespace
+
+template <typename _ExecutionPolicy, typename _Range1, typename _Range2, typename _Range3, typename _Range4,
+          typename _BinaryPredicate, typename _BinaryOperator>
+oneapi::dpl::__internal::__enable_if_hetero_execution_policy<_ExecutionPolicy,
+                                                             oneapi::dpl::__internal::__difference_t<_Range3>>
+sycl_reduce_by_segment(_ExecutionPolicy&& __exec, _Range1&& __keys, _Range2&& __values, _Range3&& __out_keys,
+                       _Range4&& __out_values, _BinaryPredicate __binary_pred, _BinaryOperator __binary_op,
+                       ::std::false_type /* has_known_identity */)
+{
+    typedef uint64_t CountType;
```

Comment:
1. just for keep one style - we use `using `instead of `typedef`.
2. uint64_t may not match to `neapi::dpl::__internal::__difference_t<_Range3>`
3. Actually, is better to keep just one line `return oneapi::dpl::experimental::ranges::reduce_by_segment.....` instead of lines  200-204.

Reply:
I have combined lines 200-204 to a single return.